src/main/java/org/candlepin/model/OwnerProductCurator.java

```diff
@@ -282,16 +318,20 @@ public long getOwnerCount(Product product) {
      */
     @Transactional
     public boolean productExists(Owner owner, String productId) {
-        String jpql = "SELECT count(op) FROM OwnerProduct op " +
-            "WHERE op.owner.id = :owner_id AND op.product.id = :product_id";
-
-        long count = (Long) this.getEntityManager()
-            .createQuery(jpql)
-            .setParameter("owner_id", owner.getId())
-            .setParameter("product_id", productId)
-            .getSingleResult();
-
-        return count > 0;
+        String jpql = "SELECT op FROM OwnerProduct op " +
+            "WHERE op.ownerId = :owner_id AND op.productId = :product_id";
+
+        try {
+            this.getEntityManager()
+                .createQuery(jpql)
+                .setParameter("owner_id", owner.getId())
+                .setParameter("product_id", productId)
+                .getSingleResult();
+        }
+        catch (NoResultException nre) {
+            return false;
+        }
+        return true;
```

Comment:
Let's rearrange the return statements so that our default output is false, and the confirmation only happens from within the try block. Largely meaningless, but helps readability for any refactors in the future and keeps it consistent with other, similar code.